docs/quickstart.md

```diff
@@ -213,6 +219,80 @@ info: Echo Handler: teardown method called.
 info: Echo Behaviour: teardown method called.
 ```
 
+## Write a test for the AEA
+
+We can write an end-to-end test for the AEA utilising helper classes provided by the framework.
+
+<details><summary>Step by step install</summary>
+
+The following test class replicates the preceding demo and tests it's correct behaviour. The `AEATestCase` is a tool for AEA developers to write useful end-to-end tests of their AEAs.
+
+``` python
+import os
+import signal
+import time
+
+from aea.connections.stub.connection import (
+    DEFAULT_INPUT_FILE_NAME,
+    DEFAULT_OUTPUT_FILE_NAME,
+)
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.test_tools.decorators import skip_test_ci
+from aea.test_tools.generic import (
+    read_envelope_from_file,
+    write_envelope_to_file,
+)
+from aea.test_tools.test_cases import AEATestCase
+
+
+class TestEchoSkill(AEATestCase):
+    """Test that echo skill works."""
+
+    @skip_test_ci
+    def test_echo(self, pytestconfig):
+        """Run the echo skill sequence."""
+        self.initialize_aea()
+        agent_name = "my_first_agent"
+        self.create_agents(agent_name)
+
+        agent_dir_path = os.path.join(self.t, agent_name)
+        os.chdir(agent_dir_path)
+
+        self.add_item("skill", "fetchai/echo:0.1.0")
+
+        process = self.run_agent()
+        time.sleep(2.0)
+
+        # add sending and receiving envelope from input/output files
+        message = DefaultMessage(
+            performative=DefaultMessage.Performative.BYTES, content=b"hello",
+        )
+        sent_envelope = Envelope(
+            to=agent_name,
+            sender="sender",
+            protocol_id=message.protocol_id,
+            message=DefaultSerializer().encode(message),
+        )
+
+        write_envelope_to_file(sent_envelope, DEFAULT_INPUT_FILE_NAME)
+
+        time.sleep(2.0)
+        received_envelope = read_envelope_from_file(DEFAULT_OUTPUT_FILE_NAME)
+
+        assert sent_envelope.to == received_envelope.sender
+        assert sent_envelope.sender == received_envelope.to
+        assert sent_envelope.protocol_id == received_envelope.protocol_id
+        assert sent_envelope.message == received_envelope.message
+
+        process.send_signal(signal.SIGINT)
+        process.wait(timeout=20)
+        assert process.returncode == 0
```

Comment:
Also this part could be improved. Put it in another method like `self.terminate_agent(signal=signal.SIGINT, timeout=20)`

Reply:
Nice! combine these suggestions in issue below?